src/test/kotlin/no/nav/tilleggsstonader/sak/vedtak/boutgifter/beregning/StepDefinitions.kt

```diff
@@ -0,0 +1,152 @@
+import no.nav.tilleggsstonader.sak.opplysninger.grunnlag.GrunnlagsdataService
+import no.nav.tilleggsstonader.sak.util.saksbehandling
+import no.nav.tilleggsstonader.sak.vedtak.TypeVedtak
+import no.nav.tilleggsstonader.sak.vedtak.VedtakRepository
+import no.nav.tilleggsstonader.sak.vedtak.boutgifter.domain.Beregningsgrunnlag
+import no.nav.tilleggsstonader.sak.vedtak.boutgifter.domain.BeregningsresultatBoutgifter
+import no.nav.tilleggsstonader.sak.vedtak.boutgifter.domain.BeregningsresultatForLøpendeMåned
+import no.nav.tilleggsstonader.sak.vedtak.domain.TypeBoutgift
+import no.nav.tilleggsstonader.sak.vedtak.domain.Vedtaksperiode
+import no.nav.tilleggsstonader.sak.vedtak.læremidler.beregning.BeregningNøkler
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.VilkårperiodeService
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.AktivitetType
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.MålgruppeType
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.Vilkårperioder
+import org.assertj.core.api.Assertions.assertThat
+import org.slf4j.LoggerFactory
+import java.util.UUID
+
+@Suppress("unused", "ktlint:standard:function-naming")
+class StepDefinitions {
+    val logger = LoggerFactory.getLogger(javaClass)
+
+    val utgiftService = mockk<BoutgifterUtgiftService>()
+
+    val vedtakRepositroy = mockk<VedtakRepository>()
+    val vilkårperiodeService = mockk<VilkårperiodeService>(relaxed = true)
+    val grunnlagsdataService = mockk<GrunnlagsdataService>(relaxed = true)
+    val vedtaksperiodeValideringService = mockk<BoutgifterVedtaksperiodeValideringService>(relaxed = true)
+
+    val beregningsService =
+        BoutgifterBeregningService(
+            boutgifterUtgiftService = utgiftService,
+            vedtaksperiodeValideringService = vedtaksperiodeValideringService,
+            vedtakRepository = vedtakRepositroy,
+        )
+
+    val behandling = saksbehandling()
+
+    var vilkårperioder: Vilkårperioder? = null
+    var vedtaksperioder: List<Vedtaksperiode> = emptyList()
+    var utgifter = mutableMapOf<TypeBoutgift, List<UtgiftBeregningBoutgifter>>()
+    var beregningsresultat: BeregningsresultatBoutgifter? = null
+    var exception: Exception? = null
+
+    @Gitt("følgende vedtaksperioder for boutgifter")
+    fun `følgende vedtaksperioder for boutgifter`(dataTable: DataTable) {
+        vedtaksperioder =
+            dataTable.mapRad { rad ->
+                Vedtaksperiode(
+                    id = UUID.randomUUID(),
+                    fom = parseDato(DomenenøkkelFelles.FOM, rad),
+                    tom = parseDato(DomenenøkkelFelles.TOM, rad),
+                    målgruppe = parseValgfriEnum<MålgruppeType>(BeregningNøkler.MÅLGRUPPE, rad) ?: MålgruppeType.AAP,
+                    aktivitet =
+                        parseValgfriEnum<AktivitetType>(BeregningNøkler.AKTIVITET, rad)
+                            ?: AktivitetType.TILTAK,
+                )
+            }
```

Comment:
Det finnes en util for dette, `mapVedtaksperioder` i `BeregningParser.kt` i tilsyn barn.
Mulig den hjelpemetoden kan flyttes på 2 steg opp og brukes her :-)